mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -79,6 +83,7 @@ public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable)
             "Iceberg table already created - can not use provided schema");
         Preconditions.checkArgument(catalogProperties.getProperty(InputFormatConfig.PARTITION_SPEC) == null,
             "Iceberg table already created - can not use provided partition specification");
+        // TODO: Check type compatibility between this.icebergTable and hmsTable.getSd().getCols()
```

Comment:
What about disallowing a schema in Hive DDL when the table already exists? Then we could always fill in the schema from the Iceberg table.

Reply:
Iceberg schema definition is sometimes richer, so a user might want to specify both the type in Hive and in Iceberg by hand and would not want to rely on the automatic conversion.

It is a non-trivial task here to find out here if the columns are generated from the SQL statement or from the Iceberg table schema so I left it as it is.
Shall we sink effort here to temporarily forbid table creation?

Reply:
Ok. Implemented the first version of the check